components/analytics/InsightPanel.js

```
@@ -265,9 +324,36 @@ export default function InsightPanel({ data, days = 30, isExpanded = false, isMa
+
+            {/* Centered spinner loader */}
+            <div className="absolute inset-0 z-10 flex items-center justify-center bg-white/40 dark:bg-slate-900/40 backdrop-blur-[2px]">
+              <div className="flex items-center gap-2 bg-white/80 dark:bg-slate-800/80 px-3 py-1.5 rounded-lg shadow-sm border border-indigo-100 dark:border-slate-700/50">
+                <Loader2 size={14} className="animate-spin text-indigo-600 dark:text-indigo-400" />
+                <span className="text-xs font-bold text-indigo-600 dark:text-indigo-400 tracking-wide">Analyzing Trends...</span>
+              </div>
+            </div>
+          </div>
+        ) : aiError ? (
+          <div className="rounded-xl border border-red-200/70 bg-red-50/70 px-4 py-3 text-sm font-medium text-red-700 dark:border-red-500/20 dark:bg-red-500/10 dark:text-red-300 flex flex-col gap-2">
+            <span className="opacity-90">We couldn't generate your trends insight right now. Please try again.</span>
```

ESLint

[error] `'` can be escaped with `&apos;`, `&lsquo;`, `&`#39`;`, `&rsquo;`.

(react/no-unescaped-entities)